comp/core/autodiscovery/autodiscoveryimpl/autoconfig.go

```diff
@@ -8,12 +8,18 @@ package autodiscoveryimpl
 
 import (
 	"context"
+	"encoding/json"
+	"net/http"
+	"sort"
 	"sync"
 	"time"
 
 	"go.uber.org/atomic"
 	"go.uber.org/fx"
 
+	"github.com/DataDog/datadog-agent/comp/api/api"
+	"github.com/DataDog/datadog-agent/comp/api/api/apiimpl/response"
```

Comment:
Why this is needed and being part of the apiimpl package? We should not import direct implementation between components

Reply:
> We should not import direct implementation between components
the current [route](https://github.com/DataDog/datadog-agent/blob/main/comp/api/api/apiimpl/internal/agent/agent.go#L401) function this is just copied from already uses [autodiscoveryimpl](https://github.com/DataDog/datadog-agent/blob/main/comp/api/api/apiimpl/internal/agent/agent.go#L408) in the api/apiimpl, I agree it is a smell but this particularly is an issue arising from missing the `<component>/defs` structure where something like this could be defined and imported separately from the impl

it looks to be the only thing in the [response/types](https://github.com/DataDog/datadog-agent/blob/main/comp/api/api/apiimpl/response/types.go#L14-L19), possibly it should be moved to the autodiscovery component? I'm not sure that is a concern for this PR but we can definitely follow up and move it

Reply:
I would support moving:

`comp/api/api/apiimpl/response` -> `comp/core/autodiscovery/common/response` or even `comp/core/autodiscovery/integration/response`

This import is currently used by:

- `cmd/cluster-agent/api/agent/agent.go`
- `pkg/flare/config_check.go`
- `pkg/snmp/snmpparse/config_snmp.go`

All of which import `autodiscovery/integration` or the `autodiscovery.Component`. I think it's fine to do so in a follow up PR, especially because this is the first of many other PRs already waiting to merge.

Reply:
@jeremy-hanna @dustmop I created the PR that move the code around https://github.com/DataDog/datadog-agent/pull/25457. That way we do not forget about it 😄